packages/decoder/lib/decoders.ts

```diff
@@ -206,14 +211,32 @@ export class ProjectDecoder {
       additionalContexts
     );
 
+    let allocations = this.allocations;
+    if (context && !(context.context in this.contexts)) {
+      //if the context comes from additionalContexts,
+      //we'll add the additional allocations to the allocations;
+      //however, we'll allow other allocations to override it...
+      //it's only supposed to be used if necessary!
+      allocations = {
+        ...this.allocations,
+        calldata: {
+          ...this.allocations.calldata,
+          functionAllocations: {
+            [context.context]: additionalAllocations,
+            ...this.allocations.calldata.functionAllocations
+          }
+        }
+      };
+    }
```

Comment:
Can the above assignment be done with
```
allocations.calldata.functionAllocations = {
  [context.context]: additionalAllocations,
  ...this.allocations.calldata.functionAllocations
};
```
Is that the same? It seems a bit simpler.

Reply:
Well, `allocations` was set to `this.allocations`, so doing it that way would alter `this.allocations, which I don't want.

Reply:
Ah right!